### ISSUE TITLE
No automatic unenrolment from unlicensed course

### ISSUE THREAD

**Issue by jlb224** [iomad/iomad#1558]
After course completion users on unlicensed courses are not being unenrolled according to enrolment expires setting. Is this expected behaviour or a bug?

Using 3.8.7, I created course as 'Users can enrol themselves onto course'
Added quiz with grade 3.0, requires passing grade
Added course completion with Requires Activity completion=quiz and Requires grade=3.0
Added course settings: 
Training expires=365
Enrolment expires=1
Days to email before=14
Days to email after=14
Days between=7
Display grade=Yes
In Assign users to course, added testuser. 
Completed course as testuser achieving passing grade on quiz.
Wait 1 day (because enrolment expires=1)
Next day: 
User not unenrolled (cron is ok)
Went to User report
Click Clear course button
User successfully unenrolled.

**Comment by turf212** [comment]
You may be suffering from the check being at today at midnight and time completed + 1 day is during the day so won't get picked up till the next day as the first check is still within 24 hours.

If there is a value in mdl_local_iomad_track for the user and the course with a timecompleted set and there is a valid for set for that course, then there will also be a value in local_iomad_track for timeexpires.  This get's set when the user completes the course. 

The task which controls all of this is the local_email_reports which runs once a day.  You can re-run the task manually and manipulate the tiemexpires to check that it is an issue with the timestamp or not.  (I.e. set the timeexpires to be timecompleted and re-run the task.  That should force it.)

**Comment by jlb224** [comment]
I have rerun the tests but still no unenrolment from the course (to note, this is the same for both licenced and unlicenced courses, not just unlicenced as in the ticket title). 

I do have values in timecompleted and timeexpires. Interestingly, timexpires is 1646225921 (which is March 2nd 2022, even though I set the enrolmentexpires to be 1) 

![image](https://user-images.githubusercontent.com/26649166/110021855-2ac7da00-7d23-11eb-9f2e-3943bd873cca.png)
![image](https://user-images.githubusercontent.com/26649166/110021888-387d5f80-7d23-11eb-8f7b-dfd0bf96f04f.png)

The only place I have 2nd March is course start date. It looks like it is using 'training expires after' (which I have set to 365) for timeexpires and is not taking into account enrolmentexpires = 1 (where it should remove 1 day after course completion)?
![image](https://user-images.githubusercontent.com/26649166/110022199-88f4bd00-7d23-11eb-90d7-90bffa56c9bb.png)
